pkg/stanza/fileconsumer/internal/tracker/tracker.go

```diff
@@ -295,3 +332,18 @@ func (t *noStateTracker) EndPoll() {}
 func (t *noStateTracker) TotalReaders() int { return 0 }
 
 func (t *noStateTracker) FindFiles([]*fingerprint.Fingerprint) []*reader.Metadata { return nil }
+
+func intToByte(val int) []byte {
+	return binary.LittleEndian.AppendUint64([]byte{}, uint64(val))
+}
+
+func byteToIndex(buf []byte) (int, error) {
+	if buf == nil {
+		return 0, nil
+	}
+	// The sizeof uint64 in binary is 8.
+	if len(buf) < 8 {
+		return 0, errInvalidValue
+	}
+	return int(binary.LittleEndian.Uint64(buf)), nil
+}
```

Comment:
Can we reuse the same marshal/unmarshal strategy from `checkpoint`? https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b6238cfe3c1988309cab87056a4bedcc0f905552/pkg/stanza/fileconsumer/internal/checkpoint/checkpoint.go#L25-L41